recipes/tensorstore/meta.yaml

```diff
@@ -0,0 +1,112 @@
+    - patches/system_nasm.patch
+    - patches/system_xz.patch
+    # Can't use keyword arguments in ctx.template()
+    - patches/repo_template.patch
+    # we are using openssl instead of boringssl (Google's fork)
+    # it mostly works out of the box, but we need to make sure the right symbol is used
+    # we are commenting out an include that doesn't exist on openssl
+    # and also replacing an old function name with its modern equivalent-ish (_cleanup->_reset)
+    - patches/oauth-boringssl.patch
+    # These two patches are FUN: we patch the workspace script from bazel
+    # so it patches the half.hpp header it will download with our meta-patch :D
+    # Idea for patch coming from # https://sourceforge.net/p/half/discussion/general/thread/86298c105c/
+    - patches/net_sourceforge_half.patch
+    - patches/half.hpp.patch.patch
+    # Use conda-forge's nlohmann_json
+    - patches/nlohmann_json-workspace.patch
+    - patches/nlohmann_json-build.patch
+    # Use conda-forge's pybind11
+    - patches/pybind11-system-build.patch
+    - patches/pybind11-workspace.patch
+    # MacOS needs an extra linker flag so pybind11 finds the python symbols in _tensorstore.so :shrug:
+    - patches/pybind11-linkopts.patch  # [osx]
+
+build:
+  skip: true  # [py<37 or win]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    # Keep bazel listed twice here to help the migrators track dependencies
+    - bazel
+    - bazel {{ bazel_version }}
+    - bazel-toolchain
+  host:
+    - pip
+    - python
+    - wheel
+    - setuptools-scm
+    - numpy
+    - libprotobuf
+    # we use openssl instead of boringssl, with some patches
+    - openssl
+    - brotli
+    - bzip2
+    - c-ares
+    - blosc
+    - libcurl
+    - libjpeg-turbo
+    - libpng
+    - lz4
+    - nasm
+    - libnghttp2
+    - snappy
+    - xz
+    - zlib
+    - zstd
+    # unvendored
+    - pybind11
+    - pybind11-abi
+    - nlohmann_json
+    # needs to be built with bazel
+    # https://github.com/abseil/abseil-cpp/blob/ee0ebda/FAQ.md#what-is-abi-and-why-dont-you-recommend-using-a-pre-compiled-version-of-abseil
+    # - abseil-cpp
+  run:
+    - {{ pin_compatible('numpy') }}
+    - python
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx and x86_64]
+
+test:
+  imports:
+    - tensorstore
+  commands:
+    - pip list
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/google/tensorstore
+  summary: Read and write large, multi-dimensional arrays
+  license: Apache-2.0 AND MIT AND BSD-3-Clause AND BSD-2-Clause
+  license_file:
+    - LICENSE
+    - licenses/com_google_absl.txt
+    - licenses/com_google_libyuv.txt
+    - licenses/com_google_re2.txt
+    - licenses/com_google_riegeli.txt
+    - licenses/net_sourceforge_half.txt
+    - licenses/org_aomedia_aom.txt
+    - licenses/org_aomedia_avif.txt
+    - licenses/org_videolan_dav1d.txt
```

Comment:
Besides absl, is there a reason for these packages to be statically linked in? AOM and libavif are available on conda-forge already.

Reply:
* `half` is headers-only and we need to patch it to make it work.
* The video libs (libyuv, aom, avif, dav1d) are all part of the same dependency tree, so maybe I can get rid of them if I patch the build files to use conda-forge dependencies, but I think they end up being statically linked anyway.
* `re2` seems to be [related to absl](https://github.com/google/tensorstore/blob/cc9a6d9b009c4fe37e5b86cb1f0087629502da5b/third_party/com_google_re2/workspace.bzl#L30)
* `riegeli` needs a [specific commit](https://github.com/google/tensorstore/blob/master/third_party/com_google_riegeli/workspace.bzl#L30) for the build

Reply:
One more note about `riegeli`, it [looks like it's designed](https://github.com/google/riegeli#status) for static linking:

> Riegeli C++ API might change in incompatible ways.

Reply:
About `re2`, they are [building off a branch named `abseil`](https://github.com/google/tensorstore/blob/cc9a6d9b009c4fe37e5b86cb1f0087629502da5b/third_party/com_google_re2/workspace.bzl#L22), which [has plenty of changes](https://github.com/google/re2/compare/main..abseil) if compared to `main`.